tests/test_target.py

```diff
@@ -14,35 +15,69 @@
 # limitations under the License.
 #
 
+""" Module for testing different targets. """
+
 import os
-import shutil
-import tempfile
 from unittest import TestCase
-
-from devlib import LocalLinuxTarget
-
+from devlib import AndroidTarget, LinuxTarget, LocalLinuxTarget
+from devlib.utils.misc import get_random_string
 
 class TestReadTreeValues(TestCase):
+    """ Class testing Target.read_tree_values_flat() """
+
+    def test_read_multiline_values(self, target):
+        """ Test Target.read_tree_values_flat() """
+        method_name = self.__class__.test_read_multiline_values.__qualname__
 
-    def test_read_multiline_values(self):
         data = {
             'test1': '1',
             'test2': '2\n\n',
             'test3': '3\n\n4\n\n',
         }
 
-        tempdir = tempfile.mkdtemp(prefix='devlib-test-')
+        dirname = os.path.join(target.working_directory,
+                               f'devlib-test-{get_random_string(12)}')
+        print(f'{method_name}: creating {dirname}...')
+        target.makedirs(dirname)
+
         for key, value in data.items():
-            path = os.path.join(tempdir, key)
-            with open(path, 'w') as wfh:
-                wfh.write(value)
+            path = os.path.join(dirname, key)
+            str_value = value.replace("\n", "\\n")
+            print(f'{method_name}: writing {str_value} to {path}...')
+            target.write_value(path, value, verify=False, as_root=target.conn.connected_as_root)
 
-        t = LocalLinuxTarget(connection_settings={'unrooted': True})
-        raw_result = t.read_tree_values_flat(tempdir)
+        print(f'{method_name}: reading values from target...')
+        raw_result = target.read_tree_values_flat(dirname)
         result = {os.path.basename(k): v for k, v in raw_result.items()}
 
-        shutil.rmtree(tempdir)
+        print(f'{method_name}: removing {dirname}...')
+        target.remove(dirname)
 
         self.assertEqual({k: v.strip()
                           for k, v in data.items()},
                          result)
+
+
+if __name__ == '__main__':
```

Comment:
I don't know if it's a good idea to leave that. These tests are supposed to be used with a test runner like `pytest`, so whatever this bit does should be done in a unittest fixture/case.

Reply:
Got it. I'll leave `test_target.py` as is and will create a new test case in another merge request.
Thanks.